### PR TITLE
[gui/launcher] first iteration of the in-game help browser and command launcher

### PR DESCRIPTION
**Pull request by myk002**
Fixes DFHack/dfhack#2224

Requires the [docs](https://github.com/DFHack/dfhack/tree/docs) branch of DFHack/dfhack.

Tab autocomplete is implemented, but the cross-indexing of commands via tags can't be (usefully) implemented until we go through and tag everything.

Enter to run the current command and show the output, shift-enter to run it and close the dialog. Dialog also auto-closes if the command brings up any screens of its own.

Screenshot:
image

### CODE REVIEW

**Comment by myk002** [comment]
we'll have to rewrite `ci/script_docs.py` to look for docs in the new location.

**Comment by myk002** [comment]
Screenshots:

Just opened:
image

After typing "auto":
image

After TABbing to autobutcher:
image

After scanning down in the help and editing the commandline:
image

After running "autobutcher list"
image

File: gui/launcher.lua [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+end
+
+----------------------------------
+-- HelpPanel
+--
+HelpPanel = defclass(HelpPanel, widgets.Panel)
+
+local DEFAULT_HELP_TEXT = [[Welcome to DFHack!
+
+Type a command to see it's help text here. Hit ENTER
```
Comment (Moth-Tolias) [inline gui/launcher.lua:263]: it's -> its

Comment (myk002) [reply gui/launcher.lua:263]: Thanks!

**Pull request merged**